In this assignment we shall use the Pandas concepts we have learnt to answer some questions about the Adults Data form the UCI ML Data Repository

In [1]:
#Importing Data and Libraries
import numpy as np
import pandas as pd

pd.set_option("display.max.columns", 100)
# to draw pictures in jupyter notebook
%matplotlib inline
# we don't like warnings
# you can comment the following 2 lines if you'd like to
import warnings

import matplotlib.pyplot as plt
import seaborn as sns

warnings.filterwarnings("ignore")


In [ ]:
DATA_URL = "https://raw.githubusercontent.com/Yorko/mlcourse.ai/main/data/"
data = pd.read_csv(DATA_URL + "adult.data.csv")

age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country salary  
0          2174             0              40  United-States  <=50K  
1             0             0              13  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40  United-States  <=50K  
4             0             0              40           Cuba  <=50K

#Q1
How many men and women (sex feature) are represented in this dataset?

In [11]:
data["sex"].value_counts()

sex
Male      21790
Female    10771
Name: count, dtype: int64

#Answer
Count of Men and Women in the Dateset
Men = 21790
Female =10771

#Q2
What is the average age (age feature) of women?

In [14]:
data[data["sex"]=="Female"]["age"].mean()

np.float64(36.85823043357163)

#Answer
The average age of all females in the dataset is 36.86 to 2 d.p

#Q3
What is the percentage of German citizens (native-country feature)?

In [32]:
data["native-country"].value_counts(normalize=True)["Germany"]*100

np.float64(0.42074874850281013)

#Answer
0.42% to 2 d.p